app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt

```diff
@@ -393,3 +436,51 @@ private fun checkIfImage(itinerary: Itinerary): Boolean {
   }
   return false // Return false if no valid URLs are found after checking all
 }
+
+/**
+ * Modifier function to add a shadow to a composable element
+ *
+ * @param color: Color of the shadow
+ * @param borderRadius: Radius of the shadow
+ * @param blurRadius: Blur radius of the shadow
+ * @param offsetY: Offset of the shadow in the y direction
+ * @param offsetX: Offset of the shadow in the x direction
+ * @param spread: Spread of the shadow
+ * @param modifier: Modifier to apply the shadow to
+ */
+fun Modifier.shadow(
+    color: Color = Color.Black,
+    borderRadius: Dp = 0.dp,
+    blurRadius: Dp = 0.dp,
+    offsetY: Dp = 0.dp,
+    offsetX: Dp = 0.dp,
+    spread: Dp = 0f.dp,
+    modifier: Modifier = Modifier
+) =
+    this.then(
+        modifier.drawBehind {
+          this.drawIntoCanvas {
+            val paint = Paint()
+            val frameworkPaint = paint.asFrameworkPaint()
+            val spreadPixel = spread.toPx()
+            val leftPixel = (0f - spreadPixel) + offsetX.toPx()
+            val topPixel = (0f - spreadPixel) + offsetY.toPx()
+            val rightPixel = (this.size.width + spreadPixel)
+            val bottomPixel = (this.size.height + spreadPixel)
+
+            if (blurRadius != 0.dp) {
+              frameworkPaint.maskFilter =
+                  (BlurMaskFilter(blurRadius.toPx(), BlurMaskFilter.Blur.NORMAL))
+            }
+
+            frameworkPaint.color = color.toArgb()
+            it.drawRoundRect(
+                left = leftPixel,
+                top = topPixel,
+                right = rightPixel,
+                bottom = bottomPixel,
+                radiusX = borderRadius.toPx(),
+                radiusY = borderRadius.toPx(),
+                paint)
+          }
+        })
```

Comment:
Clean function to get the shadow, that additionally is adjusted to the screen size. Good job there 👍